manim/utils/bezier.py

```diff
@@ -119,6 +120,56 @@ def curve(t):
     return [h0, h1, h2]
 
 
+def split_quadratic_bezier(points: Iterable[float], t: float) -> np.ndarray:
+    """
+    Split a quadratic bezier curve into two quadratic beziers at t
+
+    Parameters
+    ----------
+    points : np.array
+        The control points of the bezier curve
+        has shape [a1, h1, b1]
+
+    t : float
+        The t-value at which to split the bezier curve
+
+    Returns
+    -------
+        returns the two beziers as a list of tuples
+        has the shape [a1, h1, b1], [a2, h2, b2]
+    """
+    a1, h1, a2 = points
+    s1 = interpolate(a1, h1, t)
+    s2 = interpolate(h1, a2, t)
+    p = interpolate(s1, s2, t)
+
+    return np.array([a1, s1, p, p, s2, a2])
+
+
+def subdivide_quadratic_bezier(points: Iterable[float], n: int) -> np.ndarray:
+    """Subdivide a quadratic bezier n times
+
+    Parameters
+    ----------
+    points : np.array
+        The control points of the bezier curve in form [a1, h1, b1]
+
+    n : int
+        The number of times to subdivide the bezier curve
+
+    Returns
+    -------
+        returns the new points for the bezier curve in the form [a1, h1, b1, a2, h2, b2, ...]
+    """
+    beziers = []
+    current = points
+    for i in range(n, 0, -1):
```

Comment:
One more or less serious question: if, say, `n = 5` is passed, the `range` here is `[5, 4, 3, 2, 1]`, which splits the curve first at 1/5, then 1/4 of the remaining etc -- until `i = 1` is reached, which is when `split_quadratic_bezier(current, 1.0)` is called. Is that really helpful? Instead of having `subdivide_quadratic_bezier` construct `n` splits (and thus `n+1` subcurves, where the last split is actually just the last point), wouldn't it make more sense, semantically, to have the function construct `n-1` splits such that the curve is *subdivided* into `n` pieces?

Reply:
I might've messed up my indexing. At least I didn't intend to call split at t=1.

Very good point xD

Reply:
Wait nvm it is useful to call split with `1`. That just copies the last curve without modifying it.

```py
[[-3.00256711 -1.          0.        ]
 [-1.98967748  2.          0.        ]
 [ 2.99743289 -1.          0.        ]
 [ 2.99743289 -1.          0.        ]
 [ 2.99743289 -1.          0.        ]
 [ 2.99743289 -1.          0.        ]]
```

as you can see the last 3 points are a copy of the third point of the quadratic bezier so the first 3 correspond to the original bezier curve. That makes it easier to add the last curve without having an if statement.

I just needed to remember what i did XD

Reply:
```
class BezierSubdivision(Scene):
    def construct(self):
        a1, h1, a2 = Dot([-3, -1, 0]), Dot([0, 2, 0]), Dot([3, -1, 0])

        def update():
            return VGroup(
                *[
                    VGroup(
                        MultiBezier(
                            subdivide_bezier(
                                np.array(
                                    [a1.get_center(), h1.get_center(), a2.get_center()]
                                ),
                                i,
                            )
                        ),
                        Tex(f"n={i}"),
                    ).arrange(DOWN)
                    for i in range(1, 5)
                ]
            ).arrange_in_grid(n_rows=2, n_cols=2)

        beziers = always_redraw(update)

        self.add(beziers)
```
![BezierSubdivision_ManimCE_v0 15 2](https://user-images.githubusercontent.com/17803932/177278487-39e844c9-fb03-4e01-b7ea-5c69d6ebb3cc.png)

So it actually already does n-1 splits 👀 because splitting at 1 is a no split. I mean we could also add an if statement to have that extra. But it really depends on how much it affects the performance. 🤷🏼

Reply:
So yeah the code wasn't the issue but the comment 🤦🏼